plugins/dfu/dfu-target.c

```diff
@@ -721,6 +721,12 @@ dfu_target_download_chunk (DfuTarget *target, guint16 index, GBytes *bytes, GErr
 		g_usleep (dfu_device_get_download_timeout (priv->device) * 1000);
 	}
 
+	/* The DFU Leave operation is effectively executed. If the status returned by the device is dfuMANIFEST
+     detach has occurred, expect the device to disconnect */
+	if (dfu_device_get_state(priv->device) == DFU_STATE_DFU_MANIFEST) { // the device
+			return TRUE;
```

Comment:
whitespace is wrong for the project, please line this up.

I think that it's worth mentioning this context in a debugging line (ie g_debug ("device has detached and will disconnect")